Hacktoberfest 2019 - Move FAQ to a separate page (#2493)
jenkins-infra/jenkins.io: content/events/hacktoberfest/faq.adoc

---
layout: project
title: "Hacktoberfest in Jenkins. FAQ"
sigId: "advocacy-and-outreach"
tags:
  - outreach-programs
  - community
  - outreach
  - programs
  - hacktoberfest
links:
  gitter: jenkinsci/hacktoberfest
opengraph:
  image: /images/hacktoberfest/2019_social.png
---

NOTE: You can find Hacktoberfest FAQ link:https://hacktoberfest.digitalocean.com/faq[here].
Below you can find answer to some Jenkins-specific questions.

== Committers

This section provides answers to anyone who wants to participate in hacktoberfest and
to submit pull requests to the Jenkins project.

=== I am new to Jenkins, how do I start?

If you are new to Jenkins,
you could start by fixing some small and well described issues in the featured projects.

> There are lists of such newbie-friendly issues, see the links in the table above.

Reviewer: ```suggestion
There are lists of such newbie-friendly issues, see the links in our link:/events/hacktoberfest#featured-projects[featured project list].
```

You can also submit your own issue and propose a fix.

=== I want to work on my own plugin, is it fine?

Yes, it is fine!
Any contributions count, your role in a repository does not matter.
Just make sure you create pull requests instead of direct pushes
(hint: it's a best practice anyway if you have a CI configured for your repository).

=== How to find documentation?

Jenkins project contains lots of materials about contributing to the project.
Here are some links which may help:

* link:/participate/[Participate] - landing page for newcomer contributors
* link:/blog/2017/08/07/intro-to-plugin-development/[Plugin Development Tutorials]
* link:/doc/developer/[Developer Documentation]
* link:https://gitter.im/jenkinsci/hacktoberfest[Gitter channel] for Q&A

Projects in the table above also have their own documentation to help newcomers.

=== How do I get reviews?

> All featured projects in the list above are monitored by their maintainers,

Reviewer: ```suggestion
All featured projects are monitored by their maintainers,
```

and you will likely get a review within few days.
Reviews in other repositories and plugins may take longer.

> In the case of delays, ping us in the link:https://gitter.im/jenkinsci/hacktoberfest[hacktoberfest] channel in Gitter.

Reviewer: ```suggestion
In case of delays, ping us in the link:https://gitter.im/jenkinsci/hacktoberfest[hacktoberfest] channel in Gitter.
```

Unmerged pull-requests also count in Hacktoberfest,
so merge delays won't block you from getting prizes.

=== I am stuck. How do I get help?

* For general questions (process and general direction) use our link:https://gitter.im/jenkinsci/hacktoberfest[Hacktoberfest Gitter channel]
* You can also use other link:/chat[chats] or
link:/mailing-lists/[mailing lists].
Many subprojects also have their own chats, and we encourage using them if you want to reach out to the wider community.

=== Does Jenkins project send special swag?

> All participants will get swag from Hacktoberfest organizers if they create at least 5 pull requests.

Reviewer: ```suggestion
All participants will get swag from Hacktoberfest organizers if they create at least 4 pull requests.
```
